test-resources/api-tests/utils/apiTestHelpers.ts

```diff
@@ -0,0 +1,303 @@
+  const authorizationHeader = "x-custom-auth";
+  const accessToken = await getCredentialAccessToken(
+    PROXY_API_INSTANCE,
+    clientIdAndSecret,
+    authorizationHeader,
+  );
+  const requestBody = getCredentialRequestBody(clientDetails);
+  PROXY_API_INSTANCE.post(`/async/credential`, requestBody, {
+    headers: {
+      [authorizationHeader]: "Bearer " + accessToken,
+    },
+  });
+}
+
+export async function getActiveSessionId(sub: string): Promise<string> {
+  await createSessionForSub(sub);
+  const accessToken = await getActiveSessionAccessToken(sub);
+  const { sessionId } = (
+    await SESSIONS_API_INSTANCE.get("/async/activeSession", {
+      headers: { Authorization: `Bearer ${accessToken}` },
+    })
+  ).data;
+
+  return sessionId;
+}
+
+async function getFirstRegisteredClient(): Promise<ClientDetails> {
+  const clientsDetails = await getRegisteredClients();
+  return clientsDetails[0];
+}
+
+async function getRegisteredClients(): Promise<ClientDetails[]> {
+  const secretsManagerClient = new SecretsManagerClient({
+    region: "eu-west-2",
+  });
+  const secretName = `${process.env.TEST_ENVIRONMENT}/clientRegistryApiTest`;
+  const command = new GetSecretValueCommand({
+    SecretId: secretName,
+  });
+  const response = await secretsManagerClient.send(command);
+  return JSON.parse(response.SecretString!);
+}
+
+async function getCredentialAccessToken(
+  apiInstance: AxiosInstance,
+  clientIdAndSecret: string,
+  authorizationHeader: string,
+): Promise<string> {
+  const response = await apiInstance.post(
+    `/async/token`,
+    "grant_type=client_credentials",
+    {
+      headers: {
+        [authorizationHeader]: "Basic " + toBase64(clientIdAndSecret),
+      },
+    },
+  );
+
+  return response.data.access_token as string;
+}
+
+function toBase64(value: string): string {
+  return Buffer.from(value).toString("base64");
+}
+
+function getCredentialRequestBody(
+  clientDetails: ClientDetails,
+  sub?: UUID | undefined,
+): CredentialRequestBody {
+  return <CredentialRequestBody>{
+    sub:
+      sub ?? "urn:fdc:gov.uk:2022:56P4CMsGh_02YOlWpd8PAOI-2sVlB2nsNU7mcLZYhYw=",
+    govuk_signin_journey_id: "44444444-4444-4444-4444-444444444444",
+    client_id: clientDetails.client_id,
+    state: "testState",
+    redirect_uri: clientDetails.redirect_uri,
+  };
+}
+
+async function getActiveSessionAccessToken(
+  sub?: string,
+  scope?: string,
+): Promise<string> {
+  const requestBody = new URLSearchParams({
+    subject_token: sub ?? randomUUID(),
+    scope: scope ?? "idCheck.activeSession.read",
+    grant_type: "urn:ietf:params:oauth:grant-type:token-exchange",
+    subject_token_type: "urn:ietf:params:oauth:token-type:access_token",
+  });
+  const stsMockResponse = await STS_MOCK_API_INSTANCE.post(
+    "/token",
+    requestBody,
+  );
+
+  return stsMockResponse.data.access_token;
+}
+
+async function createSessionForSub(sub: string) {
+  const clientDetails = await getFirstRegisteredClient();
```

Comment:
nit
```suggestion
  const clientDetails = await getRegisteredClientTestData();
```

or `getRegisteredClientDetails`

i.e. I don't think we need to have the word First in there :)

Reply:
updated